README.md

###Git repository of java-assess framework is located in AFS::/p/swamp/gits/java-assess-master


###To clone the latest version from the java-assess repository:
###This would get the latest version of the java-assess into the ./java-assess-master
```sh
%% git clone <user-account>@<host-name>:/p/swamp/gits/java-assess-master
```


###To pull updates from the central repository 
```sh
%% git pull origin master
```


###To push updates from the central repository 
```sh
%% git push origin master
```


###To push tags 
```sh
%% git push origin --tags
```


###To create a release bundle (to MIR)
(
cd java-assess-master
 ./util/create_release_bundle.sh <destination directory> 
)




release/RELEASE_NOTES.txt
-----------------------------------
java-assess version 2.7.6 (2020-05-21)
-----------------------------------
- fix bug due to failure to initialize base class for
  JavaAndroidApkPkg

-----------------------------------
java-assess version 2.7.5 (2020-05-20)
-----------------------------------
- add missing keys and values to build.conf: build-stderr-file,
  build-stdout-file, config-stderr-file, config-stdout-file, and
  source-compiles 
- changed build.conf key: build-root-dir -> build-dir
- use stop-ts instead of end-ts in assessment_summary.xml
- Add prototype cryptoguard swamp tool

-----------------------------------
java-assess version 2.7.4 (2020-04-09)
-----------------------------------
- Update dependencies for newer versions of platforms
- Update java configuration for newer versions of platforms
- Eliminate version-specific ps-jtest configuration; the
  versions were not matched, which caused the tool to log
  an unneeded warning.   Now it is version independent.

-----------------------------------
java-assess version 2.7.3 (2020-01-28)
-----------------------------------
- support negative epoch seconds and negative time differences; will not occur
  in normal operation, but can if the clock is not set or goes backwards

-----------------------------------
java-assess version 2.7.2 (2020-01-16)
-----------------------------------
- Fix Parasoft J-test assessment failures when both the package and build
  directories were undefined or '.'
- Improved low-level provisioning code to include additional diagnostic
  information, and other small improvements

-----------------------------------
java-assess version 2.7.1 (2019-12-11)
-----------------------------------
- always read parsed_results_data.conf if available
- generate a proper error if parsed_results_data.conf is missing
- always get task information for parse-results from weakness_count.out file
- put stdout and stderr of the resultparser in resultparser_stdout.out, and
  resultparser_stderr.out in the parsed_results directory

-----------------------------------
java-assess version 2.7.0 (Thu Dec 05 14:00:00 CDT 2019)
-----------------------------------
- Added support to log non-assessment commands to assessment_summary
- Added execution-successful property to assessment_summary
- Fixed executable and args list in summary files
- Added support to configure tools and result parser via services.conf
- Support --version for build_assess_driver
- Support new result parser options
- This version requires resultparser-3.3.0 or later
- Improved low-level provisioning code's logging, reliability and
  support for running in both VMs and Docker containers

-----------------------------------
java-assess version 2.6.14 (Wed Oct 10 11:50:37 CDT 2018)
-----------------------------------
- new modular run.sh to allow further changes and improve reliability
- variable exports changed for run.sh changes

-----------------------------------
java-assess version 2.6.13 (Wed Aug 21 11:50:37 CDT 2018)
-----------------------------------
- allow services.conf to over-ride tool configuration

-----------------------------------
java-assess version 2.6.8 (Wed Jun 27 11:50:37 CDT 2018)
-----------------------------------
- Added -Dhttps.protocols=TLSv1.2 to build commands to fix failures caused by maven central depricating TLSv1.1

- Updated gradle to 4.8.1, as gradle does not care about the https.protocols=TLSv1.2 property

-----------------------------------
java-assess version 2.6.7 (Thu Jan 18 15:32:34 CST 2018)
-----------------------------------
- Fixed Jtest license error handling problem with 10.3.3

- Jtest configuration can now be updated from services.conf

-----------------------------------
java-assess version 2.6.6 (Thu Nov  9 15:29:12 CST 2017)
-----------------------------------
- Added support for Jtest10

- Added support for Spotbugs

-----------------------------------
java-assess version 2.6.5 (Thu Aug 31 10:40:11 CDT 2017)
-----------------------------------
- Fixed a bug with status.out:assess (pass: 0, fail: 0)

- Updated RELEASE NOTES

-----------------------------------
java-assess version 2.6.4 (Wed Aug  9 15:28:09 CDT 2017)
-----------------------------------
- Fixed dependency check backwards compatibility

-----------------------------------
java-assess version 2.6.2 (Tue Aug  8 11:25:35 CDT 2017)
-----------------------------------
- Added error handling code for dependency check

-----------------------------------
java-assess version 2.6.1 (Fri Aug  4 15:23:12 CDT 2017)
-----------------------------------
- Fixed ODC issues

-----------------------------------
java-assess version 2.6.0 (Tue Aug  1 14:35:43 CDT 2017)
-----------------------------------
- Added error handling for OWASP Depedency Check

-----------------------------------
java-assess version 2.5.8 (Tue Jul 25 08:17:32 CDT 2017)
-----------------------------------
- Handle different exit code for error-prone-2.0.21

- framework version in build summary and assessment summary

- Refactored Maven plugin

-----------------------------------
java-assess version 2.5.6 (Thu May 25 10:22:12 CDT 2017)
-----------------------------------
- Fixed Jtest bug with include file filters: If package name does not match with the directory structure

- Refactored swamp-maven-plugin and how swamp-maven-plugin is attached to the build

-----------------------------------
java-assess version 2.5.5 (Fri May 19 11:20:42 CDT 2017)
-----------------------------------
- Fixed Jtest bug with include file filters: If package name does not match with the directory structure

- Added package-name, package-version, platform-name to assessment_summary.xml

- Fixed package-root-dir

- Deleted files manually

- Add ant, maven and gradle versioning for non-androiid assessments

- Android has its own considerations

- Configure java for android sdk

- Do java7/java8 java options (they changed)

- Add java9 to some platforms

- Configure java-android-sdk for sdk-dependent java


src/gencmd_parsetab.py

# gencmd_parsetab.py
# This file is automatically generated. Do not edit.
_tabversion = '3.8'

_lr_method = 'LALR'

_lr_signature = '1E0349E66FEF050B98C84274B5005B31'
    
_lr_action_items = {'QSTRING':([0,9,17,18,21,],[7,7,7,7,7,]),'STRING':([0,9,17,18,21,],[2,2,2,2,2,]),'NEWLINE':([1,2,3,5,6,7,8,10,11,13,14,15,17,19,20,22,23,25,],[9,-11,-3,-4,-2,-12,-19,-7,-8,-9,-10,18,-13,-14,-18,-17,-16,-15,]),'SEPERATER':([17,],[21,]),'$end':([4,9,12,16,18,24,],[0,-20,-1,-6,-20,-5,]),'PARAM':([0,9,17,18,21,],[8,8,8,8,8,]),'OPTIONNAME':([9,18,],[17,17,]),}

_lr_action = {}
for _k, _v in _lr_action_items.items():
   for _x,_y in zip(_v[0],_v[1]):
      if not _x in _lr_action:  _lr_action[_x] = {}
      _lr_action[_x][_k] = _y
del _lr_action_items

_lr_goto_items = {'executable':([0,],[1,]),'optionarg':([17,21,],[19,25,]),'arg':([9,18,],[15,15,]),'option':([9,18,],[10,10,]),'string':([0,9,17,18,21,],[3,11,20,11,20,]),'command':([0,],[4,]),'args':([9,18,],[12,24,]),'quotedstring':([0,9,17,18,21,],[5,13,22,13,22,]),'param':([0,9,17,18,21,],[6,14,23,14,23,]),'empty':([9,18,],[16,16,]),}

_lr_goto = {}
for _k, _v in _lr_goto_items.items():
   for _x, _y in zip(_v[0], _v[1]):
       if not _x in _lr_goto: _lr_goto[_x] = {}
       _lr_goto[_x][_k] = _y
del _lr_goto_items
_lr_productions = [
  ("S' -> command","S'",1,None,None,None),
  ('command -> executable NEWLINE args','command',3,'p_command','gencmd.py',66),
  ('executable -> param','executable',1,'p_executable','gencmd.py',70),
  ('executable -> string','executable',1,'p_executable','gencmd.py',71),
  ('executable -> quotedstring','executable',1,'p_executable','gencmd.py',72),
  ('args -> arg NEWLINE args','args',3,'p_args','gencmd.py',77),
  ('args -> empty','args',1,'p_args_empty','gencmd.py',81),
  ('arg -> option','arg',1,'p_arg','gencmd.py',85),
  ('arg -> string','arg',1,'p_arg','gencmd.py',86),
  ('arg -> quotedstring','arg',1,'p_arg','gencmd.py',87),
  ('arg -> param','arg',1,'p_arg','gencmd.py',88),
  ('string -> STRING','string',1,'p_string','gencmd.py',93),
  ('quotedstring -> QSTRING','quotedstring',1,'p_quotedstring','gencmd.py',97),
  ('option -> OPTIONNAME','option',1,'p_option','gencmd.py',101),
  ('option -> OPTIONNAME optionarg','option',2,'p_option_value','gencmd.py',105),
  ('option -> OPTIONNAME SEPERATER optionarg','option',3,'p_option_sep_value','gencmd.py',109),
  ('optionarg -> param','optionarg',1,'p_optionarg','gencmd.py',113),
  ('optionarg -> quotedstring','optionarg',1,'p_optionarg','gencmd.py',114),
  ('optionarg -> string','optionarg',1,'p_optionarg','gencmd.py',115),
  ('param -> PARAM','param',1,'p_param','gencmd.py',120),
  ('empty -> <empty>','empty',0,'p_empty','gencmd.py',125),
]
